src/gui/transferlistwidget.cpp

```diff
@@ -59,7 +61,82 @@
 #include "transferlistsortmodel.h"
 #include "updownratiodlg.h"
 
-static QStringList extractHashes(const QList<BitTorrent::TorrentHandle *> &torrents);
+namespace
+{
+    QStringList extractHashes(const QList<BitTorrent::TorrentHandle *> &torrents)
+    {
+        QStringList hashes;
+        foreach (BitTorrent::TorrentHandle *const torrent, torrents)
+            hashes << torrent->hash();
+
+        return hashes;
+    }
+
+    // A checkable menu item that supports tristate checkboxes.
+    class CheckBoxMenuItem: public QWidgetAction
+    {
+    public:
+        using ToggleFn = std::function<void (Qt::CheckState)>;
+
+        CheckBoxMenuItem(const QString &text, const ToggleFn &onToggle, Qt::CheckState initialState, QObject *parent);
+
+    private:
+        class Widget: public QWidget
+        {
+        public:
+            Widget(const QString &text, const ToggleFn &onToggle, Qt::CheckState initialState);
+
+        protected:
+            void paintEvent(QPaintEvent *) override;
+            void mousePressEvent(QMouseEvent *) override;
+
+        private:
+            QCheckBox *const m_checkBox;
+        };
+    };
+
+    CheckBoxMenuItem::CheckBoxMenuItem(const QString &text, const ToggleFn &onToggle, Qt::CheckState initialState, QObject *parent)
+        : QWidgetAction(parent)
+    {
+        setDefaultWidget(new Widget(text, onToggle, initialState));
+    }
+
+    CheckBoxMenuItem::Widget::Widget(const QString &text, const ToggleFn &onToggle, Qt::CheckState initialState)
+        : m_checkBox(new QCheckBox(text))
+    {
+        m_checkBox->setCheckState(initialState);
+        connect(m_checkBox, &QCheckBox::stateChanged, [this, onToggle](int newState)
+                                                      {
```

Comment:
Style guide example is ambiguous about whether you are supposed to to align the braces with the left square bracket or with the start of the line. Done.